### PR TITLE
Add listenOnTCP, sendSocket,addCommand functions of console-lua-binding.

### PR DESCRIPTION
**Pull request by shujunqiao**
add listenOnTCP, sendSocket,addCommand functions of console-lua-binding and change type of command name/help.

### CODE REVIEW

**Comment by heliclei** [comment]
[ci rebuild]

**Comment by shujunqiao** [comment]
[ci rebuild]

File: cocos/base/CCConsole.h [hunk 1]
```diff
@@ -96,6 +96,8 @@ class CC_DLL Console
     void addCommand(const Command& cmd);
     /** log something in the console */
     void log(const char *buf);
+    /** send socket **/
+    void sendSocket(int fd, std::string str);
```
Comment (dumganhar) [inline cocos/base/CCConsole.h]: Please use `const std::string& str`.

File: cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp [hunk 2]
```diff
@@ -4751,6 +4751,152 @@ static void extendTMXTiledMap(lua_State* tolua_S)
...
+    tolua_error(tolua_S,"#ferror in function 'lua_cocos2dx_Console_addCommand'.",&tolua_err);
+#endif
+    
+    return 0;
+}
+
+static void extendConsole(lua_State* tolua_S)
+{
+    tolua_usertype(tolua_S,"cc.Console");
+    tolua_cclass(tolua_S,"CCConsole","cc.Console","",NULL);
```
Comment (samuele3hu) [inline cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp]: Replace "CCConsole" with "Console" would be better.

File: cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp [hunk 3]
```diff
@@ -4751,6 +4751,152 @@ static void extendTMXTiledMap(lua_State* tolua_S)
...
+    
+    tolua_beginmodule(tolua_S,"CCConsole");
+        tolua_function(tolua_S,"listenOnTCP", lua_cocos2dx_CCConsole_listenOnTCP);
+        tolua_function(tolua_S,"sendSocket", lua_cocos2dx_CCConsole_sendSocket);
+        tolua_function(tolua_S,"addCommand", lua_cocos2dx_CCConsole_addCommand);
+    tolua_endmodule(tolua_S);
+    std::string typeName = typeid(cocos2d::Console).name();
+    g_luaType[typeName] = "cc.Console";
+    g_typeCast["CCConsole"] = "cc.CCConsole";
```
Comment (samuele3hu) [inline cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp]: "cc.CCConsole" was wrong.

File: cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp [hunk 4]
```diff
@@ -4751,6 +4751,152 @@ static void extendTMXTiledMap(lua_State* tolua_S)
...
+        int arg0;
+        ok &= luaval_to_int32(tolua_S, 2,&arg0);
+        std::string arg1;
+        ok &= luaval_to_std_string(tolua_S, 3,&arg1);
+        if(!ok)
+            return 0;
+        cobj = (cocos2d::Console*)tolua_tousertype(tolua_S,1,0);
+        
+        cobj->sendSocket(arg0, arg1.c_str());
+        return 1;
```
Comment (samuele3hu) [inline cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp]: "return 1" should be "return 0".

File: cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp [hunk 5]
```diff
@@ -4751,6 +4757,173 @@ static void extendTMXTiledMap(lua_State* tolua_S)
...
+        
+        ok &= toluafix_isfunction(tolua_S,3,"LUA_FUNCTION",0,&tolua_err);
+        LUA_FUNCTION handler = 0;
+        if (ok) {
+            handler = (  toluafix_ref_function(tolua_S,3,0));
+            ScriptHandlerMgr::getInstance()->addCustomHandler((void*)cobj, handler);
+            
+            struct Console::Command outValue = {
+                name.c_str(),
+                help.c_str(),
```
Comment (dumganhar) [inline cocos/scripting/lua-bindings/manual/lua_cocos2dx_manual.cpp]: name.c_str() -> name
help.c_str() -> help

**Comment by shujunqiao** [comment]
[ci rebuild]

**Comment by shujunqiao** [comment]
[ci rebuild]

**Pull request closed without merging**